source/src/BNFC/Backend/Java.hs

```diff
@@ -97,138 +121,495 @@ makeJava options@Options{..} cf =
+            , "System.out.println();"
+            , "System.out.println(\"Parse Succesful!\");"
+            , "System.out.println();"
+            , "System.out.println(\"[Abstract Syntax]\");"
+            , "System.out.println();"
+            , "System.out.println(PrettyPrinter.show(parse_tree));"
+            , "System.out.println();"
+            , "System.out.println(\"[Linearized Tree]\");"
+            , "System.out.println();"
+            , "System.out.println(PrettyPrinter.print(parse_tree));"
+            , "return parse_tree;"
+        ]
+        , ""
+        , "public static void main(String args[]) throws Exception"
+        , codeblock 2 [
+            "Test t = new Test(args);"
+            ,"try"
+            ,codeblock 2 [
+                "t.parse();"
+            ]
+            ,"catch("<>(text err)<+>"e)"
+            ,codeblock 2 [
+                 "System.err.println(\""<>(text errmsg)<>"\");"
+                , "System.err.println(\"     \" + e.getMessage());"
+                , "System.exit(1);"
             ]
         ]
+      ]
     ]
-  where
-    eps = allEntryPoints cf
-    def = head eps
-    showOpts [] = []
-    showOpts (x:xs) | normCat x /= x = showOpts xs
-                    | otherwise      = text ('p' : identCat x) : showOpts xs
+    where
+        importfun x = "import" <+> x <> ".*;"
+        lx = text lexer
+        px = text parser
+
+        absentity = text $ (show def)
+        eps = allEntryPoints cf
+        def = head eps
+        showOpts [] = []
+        showOpts (x:xs) | normCat x /= x = showOpts xs
+                        | otherwise      = text ('p' : identCat x) : showOpts xs
+
+
+-- | Error handling in ANTLR.
+-- By default, ANTLR does not stop after any parsing error and attempts to go on,
+-- delivering what it has been able to parse. It does not throw any exception, unlike
+-- J(F)lex+CUP.
+-- To make the test class behave as with J(F)lex+CUP the additional code is needed.
+antlrErrorHandling :: String -> [Doc]
+antlrErrorHandling te = ["class"<+>tedoc<+>"extends RuntimeException"
+    ,codeblock 2 ["int line;"
+        ,"int column;"
+        ,"public"<+>tedoc<>"(String msg, int l, int c)"
+        ,codeblock 2 ["super(msg);" ,"line = l;" ,"column = c;"]]
+    ,"class BNFCErrorListener implements ANTLRErrorListener"
+    ,codeblock 2 [ "@Override"
+        ,"public void syntaxError(Recognizer<?, ?> recognizer, Object o, int i, int i1, String s, RecognitionException e)"
+        ,codeblock 2[ "throw new"<+>tedoc<>"(s,i,i1);"]
+    , "@Override"
+        ,"public void reportAmbiguity(Parser parser, DFA dfa, int i, int i1, boolean b, BitSet bitSet, ATNConfigSet atnConfigSet)"
+        ,codeblock 2[ "throw new"<+>tedoc<>"(\"Ambiguity at\",i,i1);" ]
+    ,"@Override"
+        ,"public void reportAttemptingFullContext(Parser parser, DFA dfa, int i, int i1, BitSet bitSet, ATNConfigSet atnConfigSet)"
+        ,codeblock 2 [{-"throw new"<+>tedoc<>"(\"Attempting full context\",i,i1);"-}]
```

Comment:
My comment was only concerning the last line, where you have a commented string restulting in an empty `codeblock`